java/client/src/main/java/glide/api/commands/GenericBaseCommands.java

```diff
@@ -590,4 +591,76 @@ CompletableFuture<Boolean> pexpireAt(
      * }</pre>
      */
     CompletableFuture<Boolean> copy(String source, String destination, boolean replace);
+
+    /**
+     * Serialize the value stored at <code>key</code> in a Redis-specific format and return it to the
+     * user.
+     *
+     * @see <a href="https://valkey.io/commands/dump/">valkey.io</a> for details.
+     * @param key The key of the set.
+     * @return The serialized value of a set.<br>
+     *     If <code>key</code> does not exist, <code>null</code> will be returned.
+     * @example
+     *     <pre>{@code
+     * byte[] value1 = client.dump("myKey").get();
+     * assert value1.equals("value1");
+     *
+     * byte[] value2 = client.dump("nonExistingKey").get();
+     * assert value2.equals(null);
+     * }</pre>
+     */
+    CompletableFuture<byte[]> dump(byte[] key);
+
+    /**
+     * Create a <code>key</code> associated with a <code>value</code> that is obtained by
+     * deserializing the provided serialized <code>value</code> (obtained via DUMP).
+     *
+     * @see <a href="httpshttps://valkey.io/commands/restore/">valkey.io</a> for details.
+     * @param key The key of the set.
+     * @param ttl The expiry time (in milliseconds). If 0, it means creation without any expiry.
+     * @param value The serialized value.
+     * @return Return <code>OK</code> if successfully create a <code>key</code> with a <code>value
+     *      </code>. Return a "Target key name is busy" error when <code>key</code> already exists
```

Comment:
We don't describe possible errors. I think first phrase is enough to describe the return.
```suggestion
     * @return Return <code>OK</code> if successfully create a <code>key</code> using <code>value</code>.
```